File: CourierAmexClient/src/app/features/company/models/client-category.model.ts
import { ProductModel } from "./product.model";

export interface ClientCategoryModel {
  id: number;
  companyId?: number;
  name: string;
  status: number;
  discount?: number;
  includedProducts: ProductModel[];
  excludedProducts: ProductModel[];
  companyName?: string;
}

export const newClientCategory = {
  id: 0,
  companyId: undefined,
  name: '',
  status: 2,
  includedProducts: [],
  excludedProducts: []
};

File: CourierAmexAPI/CourierAmex/Models/ClientCategoryModel.cs
namespace CourierAmex.Models
{
    public class ClientCategoryModel : BaseEntityModel<byte>
    {
        public required int CompanyId { get; set; }
        public required string Name { get; set; }
        public int Discount { get; set; }
        public IEnumerable<ProductModel>? ExcludedProducts { get; set; }
        public IEnumerable<ProductModel>? IncludedProducts { get; set; }

        public string? CompanyName { get; set; }
    }
}
